Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs

```diff
@@ -169,35 +173,98 @@ public static AuthenticationBuilder AddCustomAuthentication(this IServiceCollect
                 .AddScheme<AuthenticationSchemeOptions, CustomAuthenticationHandler>(AuthenticationSchemes.CustomAuthentication, null);
         }
 
+        /// <summary>
+        /// Sets up the proxy configuration based on the addresses in <paramref name="userList"/>.
+        /// </summary>
+        /// <param name="networkManager">The <see cref="INetworkManager"/> instance.</param>
+        /// <param name="config">The <see cref="NetworkConfiguration"/> containing the config settings.</param>
+        /// <param name="userList">The string array to parse.</param>
+        /// <param name="options">The <see cref="ForwardedHeadersOptions"/> instance.</param>
+        public static void ParseList(INetworkManager networkManager, NetworkConfiguration config, string[] userList, ForwardedHeadersOptions options)
+        {
+            for (var i = 0; i < userList.Length; i++)
+            {
+                if (IPNetAddress.TryParse(userList[i], out var addr))
```

Comment:
Couldn't the body of the `if` be extracted as a separate function and used in both ifs?

Reply:
It's a "nasty" piece of code in that it looks the same, but is different types. The first is IPNetAddresses, the second IPAddresses. It could be done, but additional complexity would need to be added.

Reply:
All you need is a function with this signature `AddIpAddress(ForwardedHeadersOptions options, IPAddress address, AddressFamily addressFamily, int prefixLength)`